nntrainer/layers/nnstreamer_layer.cpp

```diff
@@ -62,18 +62,9 @@ int NNStreamerLayer::nnst_info_to_tensor_dim(ml_tensors_info_h &out_res,
   return status;
 }
 
-NNStreamerLayer::~NNStreamerLayer() {
-  try {
-    finalizeError(ML_ERROR_NONE);
-  } catch (std::exception &e) {
-    std::cerr << "failed in destructor, reason: " << e.what();
-  }
-}
-
-void NNStreamerLayer::finalizeError(int status) {
-  if (status == ML_ERROR_NONE)
-    return;
+NNStreamerLayer::~NNStreamerLayer() { release(); }
 
+void NNStreamerLayer::release() {
```

Comment:
I guess this should be marked as noexcept (and guarantee it is noexcept), else put try catch at line 65?

Reply:
added noexcept.